mlflow/tracing/types/wrapper.py

```diff
@@ -75,20 +75,25 @@ def inputs(self) -> Dict[str, Any]:
     def outputs(self) -> Dict[str, Any]:
         return self._outputs
 
-    def set_inputs(self, inputs: Dict[str, Any]):
+    def set_inputs(self, inputs: Any):
         self._inputs = inputs
 
-    def set_outputs(self, outputs: Dict[str, Any]):
+    def set_outputs(self, outputs: Any):
         self._outputs = outputs
 
     def set_attributes(self, attributes: Dict[str, Any]):
-        try:
-            self._span.set_attributes(attributes)
-        except Exception as e:
-            _logger.warning(f"Failed to set attributes {attributes} on span {self.name}: {e}")
+        if not isinstance(attributes, dict):
+            _logger.warning(
+                f"Attributes must be a dictionary, but got {type(attributes)}. Skipping."
+            )
+            return
+        self._attributes.update(attributes)
```

Comment:
This is more like `add/update_attributes` instead of set? `set` sounds like overwriting 🤔 What's OTel's behavior?

Reply:
Yeah but OTel's `set_attributes` behavior is indeed update ([code](https://github.com/open-telemetry/opentelemetry-python/blob/721beb8b530e7a830c1e27b70c2fb9af6465baf1/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L818-L827)). I'd keep it now for consistency but can rename if we get similar feedbacks:)